src/librustdoc/externalfiles.rs

```diff
@@ -39,14 +39,32 @@ impl ExternalHtml {
         let bc = format!(
             "{}{}",
             bc,
-            Markdown(&m_bc, &[], id_map, codes, edition, playground).into_string()
+            Markdown {
+                content: &m_bc,
+                links: &[],
+                ids: id_map,
+                error_codes: codes,
+                edition,
+                playground,
+                heading_offset: HeadingOffset::H2,
```

Comment:
I must have done a bad job of explaining myself, because this is exactly backwards from what I wanted.

I was hoping for H1 on external files, and H2 on the error index, because the error index is formed by combining multiple files together, while external files are formed from one single document.

Reply:
Which reminds me that no test with only markdown and no rust code was added (completely forgot, thanks @notriddle !).

Reply:
That makes sense to me.